[
  {
    "textPayload": "Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'node-fetch' imported from /workspace/custom-pattern-endpoints.js\n    at Object.getPackageJSONURL (node:internal/modules/package_json_reader:266:9)\n    at packageResolve (node:internal/modules/esm/resolve:767:81)\n    at moduleResolve (node:internal/modules/esm/resolve:853:18)\n    at defaultResolve (node:internal/modules/esm/resolve:983:11)",
    "insertId": "69017196000c110f59c1be45",
    "resource": {
      "type": "cloud_run_revision",
      "labels": {
        "project_id": "tradiac-testing",
        "revision_name": "tradiac-api-00127-nlq",
        "service_name": "tradiac-api",
        "location": "us-central1",
        "configuration_name": "tradiac-api"
      }
    },
    "timestamp": "2025-10-29T01:44:54.790799Z",
    "severity": "ERROR",
    "labels": {
      "instanceId": "00147782966eff397b7b8d5e808ba71d3467c64dd9c33880464935cd60e9cb90f559c5fb7a985df3ff5fc188660b617789647415ab8d39d4e36c9eb51ee51d2af0c3b62e772baefbc58510fe70f4"
    },
    "logName": "projects/tradiac-testing/logs/run.googleapis.com%2Fstderr",
    "receiveTimestamp": "2025-10-29T01:44:54.796526552Z",
    "errorGroups": [
      {
        "id": "CMP5x4fwv9aDGQ"
      }
    ]
  },
  {
    "textPayload": "Default STARTUP TCP probe failed 1 time consecutively for container \"tradiac-api-1\" on port 8080. The instance was not started.\nConnection failed with status CANCELLED.",
    "insertId": "690171970004365d65811419",
    "resource": {
      "type": "cloud_run_revision",
      "labels": {
        "revision_name": "tradiac-api-00127-nlq",
        "configuration_name": "tradiac-api",
        "location": "us-central1",
        "project_id": "tradiac-testing",
        "service_name": "tradiac-api"
      }
    },
    "timestamp": "2025-10-29T01:44:55.276061Z",
    "severity": "ERROR",
    "labels": {
      "instanceId": "00147782966eff397b7b8d5e808ba71d3467c64dd9c33880464935cd60e9cb90f559c5fb7a985df3ff5fc188660b617789647415ab8d39d4e36c9eb51ee51d2af0c3b62e772baefbc58510fe70f4"
    },
    "logName": "projects/tradiac-testing/logs/run.googleapis.com%2Fvarlog%2Fsystem",
    "receiveTimestamp": "2025-10-29T01:44:55.303434495Z"
  },
  {
    "protoPayload": {
      "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
      "status": {
        "code": 9,
        "message": "Ready condition status changed to False for Revision tradiac-api-00127-nlq with message: The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start"
      },
      "serviceName": "run.googleapis.com",
      "methodName": "/Services.ReplaceService",
      "resourceName": "namespaces/tradiac-testing/revisions/tradiac-api-00127-nlq",
      "response": {
        "metadata": {
          "name": "tradiac-api-00127-nlq",
          "namespace": "941257247637",
          "selfLink": "/apis/serving.knative.dev/v1/namespaces/941257247637/revisions/tradiac-api-00127-nlq",
          "uid": "0617f10c-ddaf-4092-8187-5a53bb7bbdf7",
          "resourceVersion": "AAZCQkeCjIY",
          "generation": 1,
          "creationTimestamp": "2025-10-29T01:44:49.388289Z",
          "labels": {
            "client.knative.dev/nonce": "kbolhlbpbh",
            "serving.knative.dev/configuration": "tradiac-api",
            "serving.knative.dev/configurationGeneration": "127",
            "serving.knative.dev/service": "tradiac-api",
            "serving.knative.dev/serviceUid": "d63b335e-aeda-4f55-9b40-80daaa39a116",
            "serving.knative.dev/route": "tradiac-api",
            "cloud.googleapis.com/location": "us-central1",
            "run.googleapis.com/startupProbeType": "Default"
          },
          "annotations": {
            "autoscaling.knative.dev/maxScale": "20",
            "run.googleapis.com/client-name": "gcloud",
            "run.googleapis.com/client-version": "545.0.0",
            "run.googleapis.com/cloudsql-instances": "tradiac-testing:us-central1:tradiac-testing,tradiac-testing:us-central1:tradiac-testing-db",
            "serving.knative.dev/creator": "941257247637-compute@developer.gserviceaccount.com",
            "run.googleapis.com/build-source-location": "{\"tradiac-api-1\":\"gs://run-sources-tradiac-testing-us-central1/services/tradiac-api/1761702232.665172-2edfa17c663f420a8503e6b236ef21db.zip#1761702233365752\"}",
            "run.googleapis.com/build-id": "{\"tradiac-api-1\":\"04ee3a87-ddeb-4f80-a842-32bbcffad9d9\"}",
            "run.googleapis.com/operation-id": "19af7811-e058-49ec-ad18-33b2a2fa945a",
            "run.googleapis.com/startup-cpu-boost": "true"
          },
          "ownerReferences": [
            {
              "kind": "Configuration",
              "name": "tradiac-api",
              "uid": "46bb5472-9ac5-4197-8a9a-6ba0de2315bb",
              "apiVersion": "serving.knative.dev/v1",
              "controller": true,
              "blockOwnerDeletion": true
            }
          ]
        },
        "apiVersion": "serving.knative.dev/v1",
        "kind": "Revision",
        "spec": {
          "containerConcurrency": 80,
          "timeoutSeconds": 300,
          "serviceAccountName": "941257247637-compute@developer.gserviceaccount.com",
          "containers": [
            {
              "name": "tradiac-api-1",
              "image": "us-central1-docker.pkg.dev/tradiac-testing/cloud-run-source-deploy/tradiac-api@sha256:6d99de41b97fe7078277972a610523850f8b9270abe9adfc4f6698bc19cc40f7",
              "ports": [
                {
                  "name": "http1",
                  "containerPort": 8080
                }
              ],
              "env": [
                {
                  "name": "DB_HOST",
                  "value": "34.41.97.179"
                },
                {
                  "name": "DB_PORT",
                  "value": "5432"
                },
                {
                  "name": "DB_NAME",
                  "value": "tradiac_testing"
                },
                {
                  "name": "DB_USER",
                  "value": "postgres"
                },
                {
                  "name": "DB_PASSWORD",
                  "value": "Fu3lth3j3t!"
                }
              ],
              "resources": {
                "limits": {
                  "cpu": "1000m",
                  "memory": "512Mi"
                }
              },
              "startupProbe": {
                "timeoutSeconds": 240,
                "periodSeconds": 240,
                "failureThreshold": 1,
                "tcpSocket": {
                  "port": 8080
                }
              }
            }
          ]
        },
        "status": {
          "observedGeneration": 1,
          "conditions": [
            {
              "type": "Ready",
              "status": "False",
              "reason": "HealthCheckContainerError",
              "message": "The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start",
              "lastTransitionTime": "2025-10-29T01:44:55.276678Z"
            },
            {
              "type": "ContainerHealthy",
              "status": "False",
              "reason": "HealthCheckContainerError",
              "message": "The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start",
              "lastTransitionTime": "2025-10-29T01:44:55.276678Z"
            },
            {
              "type": "ContainerReady",
              "status": "True",
              "message": "Container image import completed in 1.27s.",
              "lastTransitionTime": "2025-10-29T01:44:51.263644Z"
            },
            {
              "type": "ResourcesAvailable",
              "status": "True",
              "message": "Provisioning imported containers completed in 1.61s. Checking container health. This will wait for up to 4m for the configured startup probe, including an initial delay of 0s.",
              "lastTransitionTime": "2025-10-29T01:44:52.879081Z"
            },
            {
              "type": "Retry",
              "status": "True",
              "reason": "ImmediateRetry",
              "message": "System will retry after 0:00:00 from lastTransitionTime for attempt 0.",
              "lastTransitionTime": "2025-10-29T01:44:52.879081Z",
              "severity": "Info"
            }
          ],
          "logUrl": "https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22",
          "imageDigest": "us-central1-docker.pkg.dev/tradiac-testing/cloud-run-source-deploy/tradiac-api@sha256:6d99de41b97fe7078277972a610523850f8b9270abe9adfc4f6698bc19cc40f7",
          "containerStatuses": [
            {
              "name": "tradiac-api-1",
              "imageDigest": "us-central1-docker.pkg.dev/tradiac-testing/cloud-run-source-deploy/tradiac-api@sha256:6d99de41b97fe7078277972a610523850f8b9270abe9adfc4f6698bc19cc40f7"
            }
          ]
        },
        "@type": "type.googleapis.com/google.cloud.run.v1.Revision"
      }
    },
    "insertId": "-lcd1osd2x1i",
    "resource": {
      "type": "cloud_run_revision",
      "labels": {
        "revision_name": "tradiac-api-00127-nlq",
        "configuration_name": "tradiac-api",
        "project_id": "tradiac-testing",
        "service_name": "tradiac-api",
        "location": "us-central1"
      }
    },
    "timestamp": "2025-10-29T01:44:55.288533Z",
    "severity": "ERROR",
    "logName": "projects/tradiac-testing/logs/cloudaudit.googleapis.com%2Fsystem_event",
    "receiveTimestamp": "2025-10-29T01:44:55.926480576Z"
  },
  {
    "protoPayload": {
      "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
      "status": {
        "code": 9,
        "message": "Ready condition status changed to False for Service tradiac-api with message: The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start"
      },
      "serviceName": "run.googleapis.com",
      "methodName": "/Services.ReplaceService",
      "resourceName": "namespaces/tradiac-testing/services/tradiac-api",
      "response": {
        "metadata": {
          "name": "tradiac-api",
          "namespace": "941257247637",
          "selfLink": "/apis/serving.knative.dev/v1/namespaces/941257247637/services/tradiac-api",
          "uid": "d63b335e-aeda-4f55-9b40-80daaa39a116",
          "resourceVersion": "AAZCQkeEkXA",
          "generation": 127,
          "creationTimestamp": "2025-10-26T15:19:06.142195Z",
          "labels": {
            "cloud.googleapis.com/location": "us-central1"
          },
          "annotations": {
            "serving.knative.dev/creator": "941257247637-compute@developer.gserviceaccount.com",
            "serving.knative.dev/lastModifier": "941257247637-compute@developer.gserviceaccount.com",
            "run.googleapis.com/build-base-image": "us-central1-docker.pkg.dev/serverless-runtimes/google-22/run/universal:public-image-next",
            "run.googleapis.com/build-enable-automatic-updates": "false",
            "run.googleapis.com/build-id": "04ee3a87-ddeb-4f80-a842-32bbcffad9d9",
            "run.googleapis.com/build-image-uri": "us-central1-docker.pkg.dev/tradiac-testing/cloud-run-source-deploy/tradiac-api",
            "run.googleapis.com/build-name": "projects/941257247637/locations/us-central1/builds/04ee3a87-ddeb-4f80-a842-32bbcffad9d9",
            "run.googleapis.com/build-source-location": "gs://run-sources-tradiac-testing-us-central1/services/tradiac-api/1761702232.665172-2edfa17c663f420a8503e6b236ef21db.zip#1761702233365752",
            "run.googleapis.com/client-name": "gcloud",
            "run.googleapis.com/client-version": "545.0.0",
            "run.googleapis.com/operation-id": "19af7811-e058-49ec-ad18-33b2a2fa945a",
            "run.googleapis.com/ingress": "all",
            "run.googleapis.com/ingress-status": "all",
            "run.googleapis.com/urls": "[\"https://tradiac-api-941257247637.us-central1.run.app\",\"https://tradiac-api-4v5l7um5lq-uc.a.run.app\"]"
          }
        },
        "apiVersion": "serving.knative.dev/v1",
        "kind": "Service",
        "spec": {
          "template": {
            "metadata": {
              "labels": {
                "client.knative.dev/nonce": "kbolhlbpbh",
                "run.googleapis.com/startupProbeType": "Default"
              },
              "annotations": {
                "autoscaling.knative.dev/maxScale": "20",
                "run.googleapis.com/client-name": "gcloud",
                "run.googleapis.com/client-version": "545.0.0",
                "run.googleapis.com/cloudsql-instances": "tradiac-testing:us-central1:tradiac-testing,tradiac-testing:us-central1:tradiac-testing-db",
                "run.googleapis.com/startup-cpu-boost": "true"
              }
            },
            "spec": {
              "containerConcurrency": 80,
              "timeoutSeconds": 300,
              "serviceAccountName": "941257247637-compute@developer.gserviceaccount.com",
              "containers": [
                {
                  "image": "us-central1-docker.pkg.dev/tradiac-testing/cloud-run-source-deploy/tradiac-api@sha256:6d99de41b97fe7078277972a610523850f8b9270abe9adfc4f6698bc19cc40f7",
                  "ports": [
                    {
                      "name": "http1",
                      "containerPort": 8080
                    }
                  ],
                  "env": [
                    {
                      "name": "DB_HOST",
                      "value": "34.41.97.179"
                    },
                    {
                      "name": "DB_PORT",
                      "value": "5432"
                    },
                    {
                      "name": "DB_NAME",
                      "value": "tradiac_testing"
                    },
                    {
                      "name": "DB_USER",
                      "value": "postgres"
                    },
                    {
                      "name": "DB_PASSWORD",
                      "value": "Fu3lth3j3t!"
                    }
                  ],
                  "resources": {
                    "limits": {
                      "cpu": "1000m",
                      "memory": "512Mi"
                    }
                  },
                  "startupProbe": {
                    "timeoutSeconds": 240,
                    "periodSeconds": 240,
                    "failureThreshold": 1,
                    "tcpSocket": {
                      "port": 8080
                    }
                  }
                }
              ]
            }
          },
          "traffic": [
            {
              "percent": 100,
              "latestRevision": true
            }
          ]
        },
        "status": {
          "observedGeneration": 127,
          "conditions": [
            {
              "type": "Ready",
              "status": "False",
              "reason": "HealthCheckContainerError",
              "message": "The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start",
              "lastTransitionTime": "2025-10-29T01:44:55.393975Z"
            },
            {
              "type": "ConfigurationsReady",
              "status": "False",
              "reason": "HealthCheckContainerError",
              "message": "The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start",
              "lastTransitionTime": "2025-10-29T01:44:55.313689Z"
            },
            {
              "type": "RoutesReady",
              "status": "False",
              "reason": "HealthCheckContainerError",
              "message": "The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable within the allocated timeout. This can happen when the container port is misconfigured or if the timeout is too short. The health check timeout can be extended. Logs for this revision might contain more information.\n\nLogs URL: https://console.cloud.google.com/logs/viewer?project=tradiac-testing&resource=cloud_run_revision/service_name/tradiac-api/revision_name/tradiac-api-00127-nlq&advancedFilter=resource.type%3D%22cloud_run_revision%22%0Aresource.labels.service_name%3D%22tradiac-api%22%0Aresource.labels.revision_name%3D%22tradiac-api-00127-nlq%22 \nFor more troubleshooting guidance, see https://cloud.google.com/run/docs/troubleshooting#container-failed-to-start",
              "lastTransitionTime": "2025-10-29T01:44:55.393975Z"
            }
          ],
          "latestReadyRevisionName": "tradiac-api-00123-j94",
          "latestCreatedRevisionName": "tradiac-api-00127-nlq",
          "traffic": [
            {
              "revisionName": "tradiac-api-00123-j94",
              "percent": 100,
              "latestRevision": true
            }
          ],
          "url": "https://tradiac-api-4v5l7um5lq-uc.a.run.app",
          "address": {
            "url": "https://tradiac-api-4v5l7um5lq-uc.a.run.app"
          }
        },
        "@type": "type.googleapis.com/google.cloud.run.v1.Service"
      }
    },
    "insertId": "-3tgnxmd5drw",
    "resource": {
      "type": "cloud_run_revision",
      "labels": {
        "service_name": "tradiac-api",
        "project_id": "tradiac-testing",
        "configuration_name": "",
        "revision_name": "",
        "location": "us-central1"
      }
    },
    "timestamp": "2025-10-29T01:44:55.424349Z",
    "severity": "ERROR",
    "logName": "projects/tradiac-testing/logs/cloudaudit.googleapis.com%2Fsystem_event",
    "receiveTimestamp": "2025-10-29T01:44:55.531080300Z"
  }
]